pkg/routine/triggertask.go

```diff
@@ -0,0 +1,90 @@
+package routine
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/iotexproject/iotex-core/pkg/lifecycle"
+)
+
+var _ lifecycle.StartStopper = (*TriggerTask)(nil)
+
+// TriggerTaskOption is option to TriggerTask.
+type TriggerTaskOption interface {
+	SetTriggerTaskOption(*TriggerTask)
+}
+
+type triggerTaskOption struct {
+	setTriggerTaskOption func(*TriggerTask)
+}
+
+func (o triggerTaskOption) SetTriggerTaskOption(t *TriggerTask) {
+	o.setTriggerTaskOption(t)
+}
+
+// WithTriggerTaskInterval sets the interval of the task
+func WithTriggerTaskInterval(d time.Duration) TriggerTaskOption {
```

Comment:
call it `AddDelayBeforeTrigger` would be more accurate? cuz it is not the `interval` of the task